### PR TITLE
Add some missing ipc message types from i3

### PR DESCRIPTION
**Pull request by ianyfan**
Namely get_binding_modes and get_config, though I'm not sure how well the latter has been implemented, would appreciate feedback on it.

Note: the config also doesn't expand includes, and I'm not sure whether that would be wanted, but it would definitely be a bit more work to add.

### CODE REVIEW

File: sway/ipc-server.c [hunk 1]
```diff
@@ -667,6 +668,31 @@ void ipc_client_handle_command(struct ipc_client *client) {
...
+		client_valid =
+			ipc_send_reply(client, json_string, (uint32_t)strlen(json_string));
+		json_object_put(modes); // free
+		goto exit_cleanup;
+	}
+
+	case IPC_GET_CONFIG:
+	{
+		json_object *json = json_object_new_object();
+		json_object_object_add(json, "config", json_object_new_string(config->current_config));
```
[the remark this answers is not in the document]
Comment (emersion) [reply sway/ipc-server.c]: Ah, this seems deliberate. The man page should be updated.

**Comment by emersion** [comment]
Can you rebase your PR please?

File: sway/ipc-server.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (RyanDwyer) [reply sway/ipc-server.c]: There is nothing different to do with JSON in `get_config` than in any other IPC command.

Literally every command in the `swaymsg` man page says it gets JSON. In most (all?) cases `swaymsg` converts the JSON into a human friendly string, unless `-r` is used. This is no different.

The question is whether the `swaymsg` docs should mention JSON in each command, when the default is to make them a human friendly string.

[the remark this answers is not in the document]
Comment (emersion) [reply sway/ipc-server.c]: Yeah, this is what confused me. This is out of scope for this PR though.

**Comment by ianyfan** [comment]
Wait, I haven't updated the ipc_feature enum, because it seems to work without it, but I still need to work out what it does.

**Comment by RyanDwyer** [comment]
That ipc_feature enum appears to be unused BTW.

**Comment by ianyfan** [comment]
Wait, don't merge yet, I missed an occurrence of current_config I need to replace

**Comment by ianyfan** [comment]
RE: ipc_feature
I can add to it but I'm not even sure what the point is if it's unused

**Comment by RyanDwyer** [comment]
Yeah, I'm fine with leaving that. Are you happy for this to be merged now?

**Comment by ianyfan** [comment]
Yes

**Comment by RyanDwyer** [comment]
Thanks!

**Pull request merged**